src/retry_op.rs

```diff
@@ -54,26 +54,23 @@ pub async fn handle_retry_error<Exe: Executor>(
             return Err(err.into());
         }
     };
-    match operation_retry_options.max_retries {
-        Some(max_retries) if current_retries < max_retries => {
-            error!(
-                "{operation_name}({topic}) answered {kind}{text}, retrying request after {:?} (max_retries = {max_retries})",
-                operation_retry_options.retry_delay
-            );
-            client
-                .executor
-                .delay(operation_retry_options.retry_delay)
-                .await;
+    if operation_retry_options.allow_retry(current_retries) {
+        let max_retries= operation_retry_options.max_retries.unwrap_or(0);
```

Comment:
```suggestion
        let max_retries = operation_retry_options.max_retries.unwrap_or(0);
```

Please run `cargo +nightly fmt --all --check` locally to make sure the format check pass

Reply:
Though I also found some existing code might be affected.

Reply:
applied.